File: Angular/src/app/shared/UserModels/Contractor.ts
import { ContractorType } from "./ContractorType";
import { User } from "./User";

export class Contractor {
contractorID: number =0;
userID : number = 0;
contractorTypeID : number = 0;
areaOfBusiness : string = "";
availability : string = "";
user : User = new User();
contractorType : ContractorType = new ContractorType();
documents : Document[] =[];
}


export class Document {
  documentID: number = 0;
  contractorID: number = 0;
  documentName: string = "";
  filePath : string ="";
  uploadDate : Date = new Date();
}

File: WebApi/Models/Users/Contractor.cs
using System.ComponentModel.DataAnnotations;
using WebApi.Models.Users;

namespace WebApi.Models;

public class Contractor
{
    public int ContractorID { get; set; }
    public int UserID { get; set; }
    public int ContractorTypeID { get; set; }
    public string AreaOfBusiness { get; set; }
    public string Availability { get; set; }
    public User User { get; set; }
    public ContractorType ContractorType { get; set; }
}

public class CreateContractorUserRequest
{
    public string Username { get; set; }
    public string Password { get; set; }
    public string ProfilePhoto { get; set; }
    public string Email { get; set; }
    public string Name { get; set; }
    public string Surname { get; set; }
    public string PhoneNumber { get; set; }
    public int ContractorTypeID { get; set; }
    public string AreaOfBusiness { get; set; }
    public string Availability { get; set; }
}

public class ContractorType
{
    [Key]
    public int ContractorTypeID { get; set; }

    [MaxLength(50)]
    public string ContractorTypeName { get; set; } = string.Empty;
}

public class ContractorTypeViewModel
{
    public string ContractorTypeName { get; set; }
}
